File: Chowdren/base/objects/backimages.cpp
// Copyright (c) Mathias Kaerlev 2012-2015.
//
// This file is part of Anaconda.
//
// Anaconda is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Anaconda is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with Anaconda.  If not, see <http://www.gnu.org/licenses/>.

#include "objects/backimages.h"
#include "collision.h"

BackImages::BackImages(int x, int y, int type_id)
: FrameObject(x, y, type_id), x_off(0), y_off(0)
{
}

int BackImages::get_width(int slot)
{
    return width;
}

class DefaultInstance : public BackImages
{
public:
    DefaultInstance()
    : BackImages(0, 0, 0)
    {
        collision = new InstanceBox(this);
        create_alterables();
        setup_default_instance(this);
    }
};

static DefaultInstance default_backimages;
FrameObject * default_backimages_instance = &default_backimages;

File: Chowdren/chowdren/writers/extensions/BackImages.py
# Copyright (c) Mathias Kaerlev 2012-2015.
#
# This file is part of Anaconda.
#
# Anaconda is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Anaconda is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Anaconda.  If not, see <http://www.gnu.org/licenses/>.

from chowdren.writers.objects import ObjectWriter

from chowdren.common import get_animation_name, to_c, make_color

from chowdren.writers.events import (ConditionMethodWriter,
    ActionMethodWriter, ExpressionMethodWriter, make_table, EmptyAction)

class BackImages(ObjectWriter):
    class_name = 'BackImages'
    filename = 'backimages'
    use_alterables = True
    default_instance = 'default_backimages_instance'

    def is_background(self):
        return False

    def write_init(self, writer):
        # XXX load x offset
        pass

actions = make_table(ActionMethodWriter, {
    6 : 'set_x',
    7 : 'set_y',
    8 : '.width = %s',
    9 : '.height = %s',
    16 : '.pattern = true',
    35 : 'set_visible(false)',
    36 : 'set_visible(true)',
    39  : 'blend_color.set_semi_transparency(%s)',
    50 : '.x_off = %s',
    51 : '.y_off = %s'
})

conditions = make_table(ConditionMethodWriter, {
})

expressions = make_table(ExpressionMethodWriter, {
    2 : 'get_width',
    3 : 'get_x()',
    4 : 'get_y()',
    5 : '.width',
    6 : '.height',
    14 : '.x_off',
    15 : '.y_off'
})

def get_object():
    return BackImages
